src/main/kotlin/com/openlattice/organizations/ExternalDatabaseManagementService.kt

```diff
@@ -271,12 +351,29 @@ class ExternalDatabaseManagementService(
         updateHBARecords(dbName)
     }
 
+    fun getOrganizationOwners(orgId: UUID): Set<SecurablePrincipal> {
+        val principals = securePrincipalsManager.getAuthorizedPrincipalsOnSecurableObject(AclKey(orgId), EnumSet.of(Permission.OWNER))
+        return securePrincipalsManager.getSecurablePrincipals(principals).toSet()
+    }
+
     /**
      * Sets privileges for a user on an organization's table or column
      */
     fun executePrivilegesUpdate(action: Action, acls: List<Acl>) {
-        val aclsByOrg = acls.groupBy { it.aclKey[0] }
-        aclsByOrg.forEach { (orgId, orgAcls) ->
+        //split acls between table and column objects
+        val aclsByType = acls.partition { it.aclKey.size == 1 }
+
+        val tableAclsByOrg = aclsByType.first.groupBy {
+            organizationExternalDatabaseTables.getValue(it.aclKey[0]).organizationId
+        }
+        val columnAclsByOrg = aclsByType.second.groupBy {
+            organizationExternalDatabaseColumns.getValue(it.aclKey[1]).organizationId
+        }
+        val orgIds = tableAclsByOrg.keys + columnAclsByOrg.keys
+
+        orgIds.forEach { orgId ->
+            val orgAcls = tableAclsByOrg[orgId]?.toMutableList() ?: mutableListOf()
+            columnAclsByOrg[orgId]?.let { orgAcls.addAll(it) }
             val dbName = PostgresDatabases.buildOrganizationDatabaseName(orgId)
             assemblerConnectionManager.connect(dbName).let { dataSource ->
                 val conn = dataSource.connection
```

Comment:
I realize this isn't your code but it also leaks db connections all over the place.

acm.connect(dbname).connection.use { conn -> 
}

and below where it does dataSource.connection.autoCommit also leaks a connection, change that to use conn also.

Reply:
Oh it's all my code lol just from previous PRs. Cleaning it all up!